### PR TITLE
Add reducer actions

### PR DESCRIPTION
**Pull request by BregornOriginal**
### In this FEATURE were added the reducer and actions for adding and deleting books.

:large_blue_diamond: Used ducks pattern to create the structure of the redux folder
:large_blue_diamond: Initial state of books is an empty array
:large_blue_diamond: Initial state of categories is an empty array
:large_blue_diamond: Use Redux Toolkit to configure the store and create actions and reducers

Technologies used: 

:small_blue_diamond: HTML
:small_blue_diamond: CSS
:small_blue_diamond: JS ES6
:small_blue_diamond: React
:small_blue_diamond: Redux
:small_blue_diamond: Redux Toolkit
:small_blue_diamond: Webpack
:small_blue_diamond: Babel
:small_blue_diamond: Jest

### CODE REVIEW

File: src/redux/bookReducer.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import { createReducer } from '@reduxjs/toolkit';
+import { addBook, deleteBook } from './books/books';
+/* eslint no-param-reassign: "error" */
+
```
Comment (Gambit142) [inline src/redux/bookReducer.js]: - [x] Kindly remove all `eslint` disable comments. `Eslint` helps us write codes that conform to industry standards and ensure that we use best practices when writing codes. Kindly remove the `eslint-disable` comment. You can read this [article](https://eslint.org/docs/latest/rules/no-param-reassign) to solve this particular `eslint` error. 👍

File: src/redux/categoriesReducer.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { createReducer } from '@reduxjs/toolkit';
+import categoriesOfBooks from './categories/categories';
+/* eslint no-param-reassign: "error" */
+
```
Comment (Gambit142) [inline src/redux/categoriesReducer.js]: - [x] Kindly remove all `eslint` disable comments. `Eslint` helps us write codes that conform to industry standards and ensure that we use best practices when writing codes. Kindly remove the `eslint-disable` comment. You can read this [article](https://eslint.org/docs/latest/rules/no-param-reassign) to solve this particular `eslint` error. 👍

File: src/redux/books/books.js [hunk 3]
```diff
@@ -0,0 +1,4 @@
+import { createAction } from '@reduxjs/toolkit';
+
+export const addBook = createAction('ADD_BOOK');
+export const deleteBook = createAction('DELETE_BOOK');
```
Comment (Gambit142) [inline src/redux/books/books.js]: - [x] Kindly use the [duck pattern](https://github.com/erikras/ducks-modular-redux) to create the value of each action type. 

#### Hint: `export const addBook = createAction('bookstore/books/ADD_BOOK')`

File: src/redux/categories/categories.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+import { createAction } from '@reduxjs/toolkit';
+
+const categoriesOfBooks = createAction('UNDER_CONSTRUCTION');
+
```
Comment (Gambit142) [inline src/redux/categories/categories.js]: - [x] Kindly use the [duck pattern](https://github.com/erikras/ducks-modular-redux) to create the value of the action type. 

#### Hint: ` const categoriesOfBooks = createAction('bookstore/categories/CHECK_STATUS')`

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @BregornOriginal 👋, 

Good job so far!

## Highlights
- `Redux toolkit` is present in `package.json` ✔️ 
- The structure of the files follows **_duck pattern_** ✔️
- Configured `Redux` store correctly ✔️
- Created `Action types`, `Action creators` and `reducers` correctly ✔️   

However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Kindly check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider

N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Review by KDlamini — APPROVED** [review]
### Hi @ 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!🎊 🥳 

## Highlights 🌞 
✔️  Configured Redux actions, reducers, and store correctly 
✔️  Good PR description
✔️  Correct use of Gitflow 
✔️  No linter errors

# Status: Approved ✔️ 

giphyapproved
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**